### PR TITLE
Make google-authenticator dependency optional

### PR DESCRIPTION
**Pull request by core23**
### Changelog

```markdown
### Changed
- Made `sonata-project/google-authenticator` an optional dependency
```

### Subject

The google-authenticator is an optional dependency now

### To do

- [X] Update composer file
- [X] Update docs

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,6 +33,7 @@
         "nelmio/api-doc-bundle": "~2.4"
     },
     "require-dev": {
+        "sonata-project/google-authenticator": "~1.0",
```
Comment (OskarStark) [inline composer.json]: not suggest?

Comment (core23) [inline composer.json]: As mentioned in https://github.com/sonata-project/SonataMediaBundle/pull/753 this should be added to **require-dev** for travis.

Comment (OskarStark) [inline composer.json]: ok, but then we need some docs how to use this in ``prod`` IMO

Comment (core23) [inline composer.json]: Added composer require to ``installation.rst``

**Comment by sagikazarmark** [comment]
:+1:

**Comment by OskarStark** [comment]
looks ok to me

ping @Soullivaneuh @rande

**Comment by core23** [comment]
@rande @Soullivaneuh Can you have a look?

**Comment by core23** [comment]
@rande Can you have a short review?

File: composer.json [hunk 2]
```diff
@@ -36,7 +35,8 @@
         "fabpot/php-cs-fixer": "~0.1|~1.0",
         "friendsofsymfony/rest-bundle": "~1.1",
         "jms/serializer-bundle": "~0.11|~1.0",
-        "nelmio/api-doc-bundle": "~2.4"
+        "nelmio/api-doc-bundle": "~2.4",
+        "sonata-project/google-authenticator": "~1.0"
```
Comment (greg0ire) [inline composer.json]: Maybe it should be added in `suggest` too?

Comment (core23) [inline composer.json]: Done :)

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by core23** [comment]
Why did you close thisn @Soullivaneuh 

IMO it's a BC break and must be done in the `master` branch?

File: composer.json [hunk 3]
```diff
@@ -36,14 +35,16 @@
...
     },
     "conflict": {
         "jms/serializer": "<0.13"
     },
     "suggest": {
         "sonata-project/doctrine-orm-admin-bundle": "~2.3",
-        "sonata-project/seo-bundle": "~1.1"
+        "sonata-project/seo-bundle": "~1.1",
+        "sonata-project/google-authenticator": "~1.0"
```
Comment (Soullivaneuh) [inline composer.json]: Please put a description and not a version.

**Comment by Soullivaneuh** [comment]
@core23 Oops, clicked too fast! :wink: 

`sonata-project/google-authenticator` should be on `conflict` section.

**Comment by sagikazarmark** [comment]
Should I close #739 in favor of this?

**Comment by Soullivaneuh** [comment]
@sagikazarmark I think yes.

File: composer.json [hunk 4]
```diff
@@ -37,16 +36,19 @@
...
-        "sonata-project/seo-bundle": "For SEO breadcrumb block service usage"
+        "sonata-project/seo-bundle": "For SEO breadcrumb block service usage",
+        "sonata-project/google-authenticator": "For google auth user login"
     },
     "conflict": {
         "jms/serializer": "<0.13",
         "sonata-project/seo-bundle": "<2.0 || >=3.0",
-        "sonata-project/doctrine-orm-admin-bundle": "<3.0 || >=4.0"
+        "sonata-project/doctrine-orm-admin-bundle": "<3.0 || >=4.0",
+        "sonata-project/google-authenticator": "<1.0"
```
Comment (Soullivaneuh) [inline composer.json]: `<1.0 || >=2.0`

**Comment by core23** [comment]
RTM? @Soullivaneuh

File: UPGRADE-4.0.md [hunk 5]
```diff
@@ -8,3 +8,7 @@ All the deprecated code introduced on 3.x is removed on 4.0.
 Please read [3.x](https://github.com/sonata-project/SonataUserBundle/tree/3.x) upgrade guides for more information.
 
 See also the [diff code](https://github.com/sonata-project/SonataUserBundle/compare/3.x...4.0.0).
+
+## Google Auth
```
Comment (OskarStark) [inline UPGRADE-4.0.md]: `Google Auth` -> `Google Authenticator`

**Comment by Soullivaneuh** [comment]
Ok for me. :+1: 

Feel free to merge when Travis will be green.

**Pull request merged**